CriticalMass/NibProviding.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  NibProviding.swift
+//  CriticalMaps
+//
+//  Created by Malte Bünz on 19.04.19.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import UIKit
+
+protocol NibProviding {
+    static var typeName: String { get }
+    static var nib: UINib { get }
+}
+
+extension NibProviding where Self: NSObject {
```

Comment:
Looks great! Maybe we could even add an extension for tableView that does the registration for a NibProviding

Reply:
yes. we could do something like that I think. But I would suggest we add that in another PR and then change it in the whole codebase